apps/src/lab2/levelEditors/neighborhood/NeighborhoodGridGenerator.tsx

```diff
@@ -0,0 +1,213 @@
+              If you want to put the same asset on multiple cells, you just need
+              to click the asset once, then click the different cells to place
+              it. If you want to have no asset selected, click the Unselect
+              Asset button.
+            </li>
+            <li>
+              When you are done, click Save Grid to save your updated grid to
+              the serialized maze field. The field will not update until you
+              click save!
+            </li>
+          </ol>
+        </div>
+      </CollapsibleSection>
+      <div className={moduleStyles.gridGenerator}>
+        <div
+          className={moduleStyles.gridContainer}
+          style={{
+            gridTemplateColumns: `repeat(${gridDimension}, 1fr)`,
+            gridTemplateRows: `repeat(${gridDimension}, 1fr)`,
+          }}
+        >
+          {grid.map((row, rowIndex) =>
+            row.map((cell, columnIndex) => {
+              const isSelected =
+                (selectedCell &&
+                  selectedCell[0] === rowIndex &&
+                  selectedCell[1] === columnIndex) ||
+                false;
+              return (
+                <img
+                  src={imageTiles[cell.assetId]}
+                  alt="neighborhood cell"
+                  width={cellSize}
+                  height={cellSize}
+                  className={isSelected ? moduleStyles.selectedCell : undefined}
+                  onClick={() => updateCell(rowIndex, columnIndex)}
+                  key={`cell-${rowIndex}-${columnIndex}`}
+                />
+              );
+            })
+          )}
+        </div>
+        <div>
+          <SimpleDropdown
+            items={categoryOptions}
+            onChange={updateCategory}
+            labelText={'Category'}
+            name={'category'}
+            className={moduleStyles.categoryDropdown}
+          />
+          <div
+            className={moduleStyles.categoryTiles}
+            style={{
+              gridTemplateRows: `repeat(${categoryRows}, 1fr)`,
+              gridTemplateColumns: `repeat(${categoryColumns}, 1fr)`,
+            }}
+          >
+            {categoryTiles.map((tile, index) => (
+              <img
+                src={imageTiles[tile]}
+                alt="neighborhood tile"
+                onClick={() => setSelectedAsset(tile)}
+                className={
+                  selectedAsset === tile ? moduleStyles.selectedTile : undefined
+                }
+                key={`tile-${index}`}
+              />
+            ))}
+          </div>
+          <Button
+            text="Unselect Asset"
+            onClick={() => setSelectedAsset(undefined)}
+            type={'secondary'}
+            color={'black'}
+            className={moduleStyles.unselectButton}
+            size={'xs'}
+            disabled={selectedAsset === undefined}
+          />
+        </div>
+      </div>
+      {selectedPaintAmount !== undefined && (
```

Comment:
Maybe later we can display counter on paint can similar to https://github.com/code-dot-org/maze/blob/9bd887d0454246dfda1f992393161185e8415559/src/drawer.js#L156

Reply:
Yeah that was out of scope for this work, but may be possible. The view here is just a bunch of images placed into a grid, not an svg, so it would be more complicated to show the number on top.